### PR TITLE
Route 53 ALIAS Target Support

### PR DESCRIPTION
**Pull request by jwaldrip**
@mitchellh I was hoping I could sneak this into 0.4.0 as well. Any chance we could make that happen?

### CODE REVIEW

**Comment by jwaldrip** [comment]
Running the acceptance tests locally right now.

**Comment by phinze** [comment]
Sorry @jwaldrip - the 0.4.0 train is already on its way. The release cycle for 0.4.1 and 0.5.0 will be much faster, so this shouldn't have to hang out too long.

**Comment by jwaldrip** [comment]
Great. I was worried It was going to be another 6 months :-).

File: builtin/providers/aws/resource_aws_route53_alias_target.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"target_zone_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Default:  "${self.zone_id}",
```
Comment (jwaldrip) [inline builtin/providers/aws/resource_aws_route53_alias_target.go]: @phinze any idea how I can reference another variable as the default if its not provided?

**Comment by jwaldrip** [comment]
@phinze how does this implementation look?

**Comment by jwaldrip** [comment]
@catsby I know you have been thinking out this... But do you have any followup comments, or is this something you want to move forward with?

**Comment by catsby** [comment]
Sorry for the lag there. Can we review why this is a separate resource, instead of along side a normal Route 53 Record resource? What are the hurdles to maintaining them as a single resource?

**Comment by jwaldrip** [comment]
Not so much a hurdle. Just some additional conditional logic. AWS handles this kind of strange, as ALIAS targets are somewhat of a sub resource of a record and invalidates many fields if enabled. Mainly, Alias targets conflict with the fields "TTL" and "records". We could make it work, but they were "different enough" in my mind to separate them.

**Comment by catsby** [comment]
Sorry for the pain here @jwaldrip – we've actually since migrated back to https://github.com/awslabs/aws-sdk-go from our fork for the AWS library, can you swap that library out? It's preventing the tests from passing :frowning: 

Let me know if you have questions on that or can't get to it, I can try and get to it this week

**Comment by radeksimko** [comment]
@catsby see https://github.com/hashicorp/terraform/pull/1775

**Comment by mitchellh** [comment]
Lets go with #1775 due to the refactor. Sorry for the delay @jwaldrip but this will get into 0.5! :)

**Comment by jwaldrip** [comment]
Good news!

**Pull request closed without merging**